estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,58 @@
+from odoo import api, fields, models
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+    _order = "price desc"
+    _check_offer_price = models.Constraint(
+        'CHECK(price > 0)',
+        'The offer price should be stricly positive'
+    )
+
+    price = fields.Float()
+    status = fields.Selection(
+        string="Status",
+        copy=False,
+        selection=[
+            ('accepted', 'Accepted'),
+            ('refused', 'Refused')
+        ]
+    )
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.add(fields.Date.today(), days=record.validity)
+
+    @api.onchange("date_deadline")
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - fields.Date.today()).days
+
+    def action_accept_offer(self):
+        for record in self:
+            if record.status == 'accepted':
+                return False
```

Comment:
It's weird to have a method that returns False in the middle of the loop.
This will lead to inconsistencies.